tests/extras/datasets/api/test_api_dataset.py

```diff
@@ -76,33 +80,75 @@ def test_http_error(self, requests_mocker, method):
         with pytest.raises(DataSetError, match="Failed to fetch data"):
             api_data_set.load()
 
-    def test_socket_error(self, requests_mocker, method):
+    def test_socket_error(self, requests_mock):
         api_data_set = APIDataSet(
-            url=TEST_URL, method=method, params=TEST_PARAMS, headers=TEST_HEADERS
+            url=TEST_URL,
+            method=TEST_METHOD,
+            load_args={"params": TEST_PARAMS, "headers": TEST_HEADERS},
         )
-        requests_mocker.register_uri(method, TEST_URL_WITH_PARAMS, exc=socket.error)
+        requests_mock.register_uri(TEST_METHOD, TEST_URL_WITH_PARAMS, exc=socket.error)
 
         with pytest.raises(DataSetError, match="Failed to connect"):
             api_data_set.load()
 
-    def test_read_only_mode(self, method):
+    def test_read_only_mode(self):
         """
         Saving is disabled on the data set.
         """
-        api_data_set = APIDataSet(url=TEST_URL, method=method)
+        api_data_set = APIDataSet(url=TEST_URL, method=TEST_METHOD)
         with pytest.raises(DataSetError, match="is a read only data set type"):
             api_data_set.save({})
 
-    def test_exists_http_error(self, requests_mocker, method):
+    def test_credentials_auth_error(self):
+        """
+        If ``auth`` in ``load_args`` and ``credentials`` are both provided,
+        the constructor should raise a ValueError.
+        """
+        with pytest.raises(ValueError, match="both auth and credentials"):
+            APIDataSet(
+                url=TEST_URL, method=TEST_METHOD, load_args={"auth": []}, credentials={}
+            )
+
+    @staticmethod
+    def _basic_auth(username, password):
+        encoded = base64.b64encode(f"{username}:{password}".encode("latin-1"))
+        return f"Basic {encoded.decode('latin-1')}"
+
+    @pytest.mark.parametrize(
+        "auth_kwarg",
+        [{"load_args": {"auth": ("john", "doe")}}, {"credentials": ("john", "doe")}],
+    )
+    def test_auth_sequence(self, requests_mock, auth_kwarg):
+        """
+        ``auth`` and ``credentials`` should be able to be any Iterable.
```

Comment:
This isn't quite right. `auth` now only works with `tuple`: if you do `"auth": ["john", "doe"]` I think this will fail. Which is ok behaviour I think, just the comment is wrong. Probably it's only worth testing the `credentials` case on this test to be honest.

Reply:
You are right, this does fail indeed. Thanks for pointing this out. I have added tests to make sure this does not happen.